MIGRATING.md

```diff
@@ -0,0 +1,242 @@
+# Migrating
+
+This guide lists API changes between *cw-plus* major releases.
+
+## v0.10.3 -> v0.11.0
+
+### Breaking Issues / PRs
+
+- Rename cw0 to utils [\#471](https://github.com/CosmWasm/cw-plus/issues/471) / Cw0 rename [\#508](https://github.com/CosmWasm/cw-plus/pull/508)
+
+The `cw0` package was renamed to `utils`. The required changes are straightforward:
+
+```diff
+diff --git a/contracts/cw1-subkeys/Cargo.toml b/contracts/cw1-subkeys/Cargo.toml
+index 1924b655..37af477d 100644
+--- a/contracts/cw1-subkeys/Cargo.toml
++++ b/contracts/cw1-subkeys/Cargo.toml
+-cw0 = { path = "../../packages/cw0", version = "0.10.3" }
++utils = { path = "../../packages/utils", version = "0.10.3" }
+```
+
+```diff
+diff --git a/contracts/cw1-subkeys/src/contract.rs b/contracts/cw1-subkeys/src/contract.rs
+index b4852225..f20a65ec 100644
+--- a/contracts/cw1-subkeys/src/contract.rs
++++ b/contracts/cw1-subkeys/src/contract.rs
+-use cw0::Expiration;
++use utils::Expiration;
+```
+
+- Deprecate `range` to `range_raw` [\#460](https://github.com/CosmWasm/cw-plus/issues/460) /
+`range` to `range raw` [\#576](https://github.com/CosmWasm/cw-plus/pull/576).
+
+`range` was renamed to `range_raw` (no key deserialization), and `range_de` to `range`. This means you can now count
+on the key to be deserialized for you, which results in cleaner / simpler code.
+
+There are some examples in the contracts code base, by example:
+```diff
+diff --git a/contracts/cw3-fixed-multisig/src/contract.rs b/contracts/cw3-fixed-multisig/src/contract.rs
+index 48a60083..2f2ef70d 100644
+--- a/contracts/cw3-fixed-multisig/src/contract.rs
++++ b/contracts/cw3-fixed-multisig/src/contract.rs
+@@ -385,21 +384,20 @@ fn list_votes(
+     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+     let start = start_after.map(Bound::exclusive);
+
+-    let votes: StdResult<Vec<_>> = BALLOTS
++    let votes = BALLOTS
+         .prefix(proposal_id)
+-        .range_raw(deps.storage, start, None, Order::Ascending)
++        .range(deps.storage, start, None, Order::Ascending)
+         .take(limit)
+         .map(|item| {
+-            let (key, ballot) = item?;
+-            Ok(VoteInfo {
+-                voter: String::from_utf8(key)?,
++            item.map(|(addr, ballot)| VoteInfo {
++                voter: addr.into(),
+                 vote: ballot.vote,
+                 weight: ballot.weight,
+             })
+         })
+-        .collect();
++        .collect::<StdResult<_>>()?;
+
+-    Ok(VoteListResponse { votes: votes? })
++    Ok(VoteListResponse { votes })
+ }
+
+ fn query_voter(deps: Deps, voter: String) -> StdResult<VoterResponse> {
+```
+
+If / when you don't need key deserialization, just can just rename `range` to `range_raw` and you are good to go.
+
+If you need / want key deserialization for **indexes**, you need to specify the primary key type as a last (optional)
```

Comment:
nit: I think "When you don't.../if you want..." variant is good, no need for "if/when ... need/want".